### PR TITLE
Fix card footer detection

### PR DESCRIPTION
**Pull request by ryantmer**
The previous implementation to show or hide the footer padding relied on `slotchange`, which I _believe_ is a Polymer 2 thing. If not, it only seemed to actually work in the Polymer 2 demo... so instead, here is a Polymer-1-compliant version of that!

### CODE REVIEW

**Comment by CodeBaboon** [comment]
@ryantmer What about inverting the logic and having the default style to include the padding and then a style that checks `:empty` to remove the padding when there are no children? https://developer.mozilla.org/en-US/docs/Web/CSS/:empty

**Comment by ryantmer** [comment]
RIGHT, `:empty`! I figured there must have been a better way to do this - not sure how `:empty` will work the slot, but I'll try!

**Comment by ryantmer** [comment]
Aw - I don't think `:empty` won't work after all. If there's no content, then the `.d2l-card-footer` div isn't empty (there's a `<slot>` in it), so the selector `.d2l-card-footer:empty` doesn't affect anything. Using the `.d2l-card-footer` class on the `<slot>` element instead and using the same selector _does_ work, but only for no content - once there's content, there's no padding, since the style is applied to the now-replaced `<slot>` element.

However, I did find a `::slotted` selector that works - there could be edge cases where it doesn't work as intended, but in the test cases that I poked around with, things seem to work :)

**Comment by dlockhart** [comment]
Boo... was really hoping `:empty` on the parent container would have worked.

**Comment by ryantmer** [comment]
Yeah - the issue was effectively that two selectors needed to be aware of each other; if there's content, we need to apply the padding to the content, but if there isn't content, we need to apply not-padding to the slot instead.

**Comment by dbatiste** [comment]
I have a wack of work planned for this component as it's just bare-bones currently.  I'll likely take another look at this as layout is adjusted.

**Comment by ryantmer** [comment]
Followed up on Slack - this is somewhat blocking, so going to merge for the time being. Better changes/solutions will certainly come over time!

File: d2l-card.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
 			.d2l-card-content {
 				padding: 1.5rem 1.2rem;
 			}
-			.d2l-card-footer[has-footer] {
+			::slotted([slot=footer]) {
```
Comment (m6jones) [inline d2l-card.html]: You can try `div ::slotted([slot=footer]) {`

Comment (ryantmer) [reply d2l-card.html]: More specific is better, good point!

**Comment by ryantmer** [comment]
Ugh - logged in to Sauce and manually ran the tests on Safari, which pass. Not sure what's up with the Travis integration, but going to merge anyway since this is fairly blocking.
